### PR TITLE
STSMACOM-614: Update internal state in SASQ when sort changes

### PR DESCRIPTION
**Pull request by mkuklis**
Update internal state in SASQ when sort changes.

https://issues.folio.org/browse/STSMACOM-614

### CODE REVIEW

File: lib/SearchAndSort/SearchAndSortQuery.js [hunk 1]
```diff
@@ -439,9 +439,15 @@ class SearchAndSortQuery extends React.Component {
...
+    const sortFields = { sort: sortOrder };
 
-    // this.log('action', `sorted by ${sortOrder}`);
-    this.callQuerySetter({ sort: sortOrder });
+    this.internalSetState(
+      { sortFields },
+      'sort.change',
+      [
+        () => this.callQuerySetter(sortFields)
+      ]
```
Comment (zburke) [inline lib/SearchAndSort/SearchAndSortQuery.js:449]: Do we need the `sortChangeCallback` here? e.g. 
```
      [
        sortChangeCallback,
        () => this.callQuerySetter(sortFields)
      ]
```

**Review by zburke — COMMENTED** [review]
This component is pretty opaque to me; I'm basing this comment based on parallel behavior in what I think are parallel methods, but the maze of clear/reset/change methods makes this hard for me to follow :/

File: lib/SearchAndSort/SearchAndSortQuery.js (hunk 1 above)
Comment (mkuklis) [reply lib/SearchAndSort/SearchAndSortQuery.js:449]: @zburke good catch! It will add it there.

**Review by JohnC-80 — APPROVED** [review]

**Review by zburke — APPROVED** [review]
Sweet! I put this branch in a `resolutions` entry in your [`ui-users` PR](/folio-org/ui-users/pull/1967) and pointed it at bugfest like 
```
yarn stripes serve --okapi https://okapi-bugfest-kiwi.folio.ebsco.com --tenant fs09000000
```
and all is well. Nice work!

**Pull request merged**